OrleansDashboard/Metrics/TypeFormatting/TypeFormatter.cs

```diff
@@ -1,56 +1,32 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.Caching;
 using System.Text;
 
-namespace OrleansDashboard
+namespace OrleansDashboard.Metrics.TypeFormatting
 {
     /// <summary>
     /// Naive parser which makes strings containing type information easier to read
     /// </summary>
     public class TypeFormatter
     {
-        enum ParseState
-        {
-            TypeNameSection,
-            GenericCount,
-            GenericArray,
-            TypeArray
-        }
+        static MemoryCache cache = MemoryCache.Default;
```

Comment:
I would use a concurrent dictionary. There is no need to ever remove something from the cache as the number of entries is relatively small.

Reply:
right :) fixed